contracts/optimistic-ethereum/OVM/bridge/messaging/Abs_BaseCrossDomainMessenger.sol

```diff
@@ -33,7 +41,9 @@ abstract contract Abs_BaseCrossDomainMessenger is iAbs_BaseCrossDomainMessenger,
      * Public Functions *
      ********************/
 
-    constructor() Lib_ReentrancyGuard() {}
+    constructor() Lib_ReentrancyGuard() internal {
+        xDomainMessageSender = DEFAULT_XDOMAIN_SENDER;
```

Comment:
Possible devex improvement here would be to make `xDomainMessageSender()` revert if the sender is this default value. Not sure how much it matters, but I feel like contracts shouldn't be able to use `xDomainMessageSender()` if this contract isn't trying to send a message during that tx.

Reply:
https://github.com/ethereum-optimism/contracts/pull/320/commits/83cff29d364f3f85d496f1aa3ca64d669903de0d what do you think about this? It requires that we remove the tests, since the test happens before/after the call and that means it always reverts.